avocado/utils/process.py

```diff
@@ -209,36 +209,41 @@ def kill_process_tree(pid, sig=signal.SIGKILL, send_sigcont=True,
 
     :param pid: The pid of the process to signal.
     :param sig: The signal to send to the processes.
+    :param send_sigcont: Send SIGCONT to allow killing stopped processes
     :param timeout: How long to wait for the pid(s) to die
                     (negative=infinity, 0=don't wait,
                     positive=number_of_seconds)
+    :return: killed_pids that were destroyed
     """
-    if timeout <= 0:
-        def remaining_time():
-            return timeout
-    else:
-        start = time.time()
+    def _all_pids_dead(killed_pids):
+        for pid in killed_pids:
+            if pid_exists(pid):
+                return False
+        return True
 
-        def remaining_time():
-            return timeout - time.time() + start
+    killed_pids = [pid]
```

Comment:
I understand the optimization here, but the name of this variable is misleading: up to this point, we cannot tell if any process has been killed -- clearly none was killed on our behalf.

Something like this:

```diff
diff --git a/avocado/utils/process.py b/avocado/utils/process.py
index 5296ef4a..f5029d8e 100644
--- a/avocado/utils/process.py
+++ b/avocado/utils/process.py
@@ -221,12 +221,12 @@ def kill_process_tree(pid, sig=signal.SIGKILL, send_sigcont=True,
                 return False
         return True
 
-    killed_pids = [pid]
     if timeout > 0:
         start = time.time()
 
     if not safe_kill(pid, signal.SIGSTOP):
-        return killed_pids
+        return [pid]
+    killed_pids = [pid]
     for child in get_children_pids(pid):
         killed_pids.extend(kill_process_tree(int(child), sig, False))
     safe_kill(pid, sig)
```

Makes reading the code straightforward IMO.

Reply:
ok